src/events/guildMemberAdd.js

```diff
@@ -1,69 +1,66 @@
-const Discord = require('discord.js');
+export const name = 'guildMemberAdd';
+export const once = false;
 
-module.exports = {
-	name: 'guildMemberAdd',
-	once: false,
-	async execute(member, client) {
-		try {
-			let guild = member.guild;
-			let guildData = await client.database.fetchGuild(guild.id); // Get guild document from database
-			if (!guildData.addons.welcome.enabled) return; // Welcome messages aren't enabled
+export async function execute(member, client) {
+    try {
+        let guild = member.guild;
+        let guildData = await client.database.fetchGuild(guild.id); // Get guild document from database
+        if (!guildData.addons.welcome.enabled) return; // Welcome messages aren't enabled
 
-			let welcomeChannel = await client.tools.resolveChannel(
-				guildData.addons.welcome.channel,
-				guild
-			); // Try find the welcome channel
+        let welcomeChannel = await client.tools.resolveChannel(
+            guildData.addons.welcome.channel,
+            guild
+        ); // Try find the welcome channel
 
-			let welcomeTitle =
-				guildData.addons.welcome.title === null ||
-				guildData.addons.welcome.title === '' ||
-				guildData.addons.welcome.title === ' '
-					? '** {user.name} ** is now a HeartStopper and member number **{guild.totalUser}**'
-					: guildData.addons.welcome.title; // Get the custom title or use the preset one
-			let welcomeMsg =
-				guildData.addons.welcome.message === null ||
-				guildData.addons.welcome.message === '' ||
-				guildData.addons.welcome.message === ' '
-					? 'Welcome {user.ping} to {guild.name}!'
-					: guildData.addons.welcome.message; // Get the custom message or use the preset one
+        let welcomeTitle =
+            guildData.addons.welcome.title === null ||
+            guildData.addons.welcome.title === '' ||
+            guildData.addons.welcome.title === ' '
+                ? '** {user.name} ** is now a HeartStopper and member number **{guild.totalUser}**'
+                : guildData.addons.welcome.title; // Get the custom title or use the preset one
+        let welcomeMsg =
+            guildData.addons.welcome.message === null ||
+            guildData.addons.welcome.message === '' ||
+            guildData.addons.welcome.message === ' '
+                ? 'Welcome {user.ping} to {guild.name}!'
+                : guildData.addons.welcome.message; // Get the custom message or use the preset one
 
-			// Replace all valid tags
-			let finalTitle = await welcomeTitle
-				.replace(/{user.ping}/g, `${member.user}`)
-				.replace(/{user.name}/g, `${member.user.username}`)
-				.replace(/{user.id}/g, `${member.user.id}`)
-				.replace(/{user.tag}/g, `${member.user.tag}`)
-				.replace(/{guild.name}/g, `${guild.name}`)
-				.replace(/{guild.id}/g, `${guild.id}`)
-				.replace(/{guild.totalUser}/g, `${guild.memberCount}`);
+        // Replace all valid tags
+        let finalTitle = welcomeTitle
```

Comment:
If we are deduplicating, would it make sense to move these replace actions into an own function?

Reply:
I tried to touch these files as little as possible to keep the impact of this PR sort-of contained. I do not oversee this file and we dont have coverage :/

Reply:
I tried to touch these files as little as possible to keep the impact of this PR sort-of contained. I do not oversee this file and we dont have coverage :/

Reply:
Perhaps a new issue would be in place for this.

Reply:
#21 covers this well enough